front-end/src/components/TranscriptUploadModal.jsx

```diff
@@ -9,22 +9,38 @@ class TranscriptUploadModal extends React.Component {
     super(props);
     this.state = {
       file: null,
+      isFileValid: null,
     };
   }
 
-  handleFile(e) {
-    this.setState({file: e.target.files})
+  async handleFile(e) {
+    let fileUploaded = new FileReader();
+    fileUploaded.readAsText(e.target.files[0]);
+    fileUploaded.onloadend = (e) => this.handleData(e.target.result);
   }
 
-  async handleUpload(e) {
-    let fileUploaded = new FileReader();
-    fileUploaded.readAsText(this.state.file[0]);
-    let result;
-    fileUploaded.addEventListener("loadend", e => {
-      result = e.target.result;
-      const formData = {transcript: result};
-      TranscriptAPI.post(formData);
-    });
+  handleData(result) {
+    if (!this.validateData(result)) {
+      this.setState({isFileValid: false});
+    } else{
+      this.setState({
+        file: {transcript: result},
+        isFileValid: true,
+      });
+    }
+  }
+
+  validateData(data) {
+    try {
+      const parsedData = JSON.parse(data);
+      return parsedData.data;
+    } catch (error) {
+      return false;
+    }
+  }
```

Comment:
Is this going to impact the rest of the formatting for backend? Like needing less parsing etc.

Reply:
It has no effect on BE.